app/src/main/java/org/schabi/newpipe/local/feed/notifications/NotificationWorker.kt

```diff
@@ -0,0 +1,124 @@
+package org.schabi.newpipe.local.feed.notifications
+
+import android.content.Context
+import androidx.core.app.NotificationCompat
+import androidx.preference.PreferenceManager
+import androidx.work.BackoffPolicy
+import androidx.work.Constraints
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.ForegroundInfo
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.PeriodicWorkRequest
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import androidx.work.rxjava3.RxWorker
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.core.Single
+import org.schabi.newpipe.R
+import org.schabi.newpipe.local.feed.service.FeedLoadManager
+import org.schabi.newpipe.local.feed.service.FeedLoadService
+import java.util.concurrent.TimeUnit
+
+/*
+ * Worker which checks for new streams of subscribed channels
+ * in intervals which can be set by the user in the settings.
+ */
+class NotificationWorker(
+    appContext: Context,
+    workerParams: WorkerParameters,
+) : RxWorker(appContext, workerParams) {
+
+    private val notificationHelper by lazy {
+        NotificationHelper(appContext)
+    }
+    private val feedLoadManager = FeedLoadManager(appContext)
+
+    override fun createWork(): Single<Result> = if (isEnabled(applicationContext)) {
+        feedLoadManager.startLoading(
+            ignoreOutdatedThreshold = true,
+            groupId = FeedLoadManager.GROUP_NOTIFICATION_ENABLED
+        )
+            .map { feed ->
+                feed.mapNotNull { x ->
+                    x.value?.takeIf {
+                        it.newStreamsCount > 0
+                    }
+                }
```

Comment:
*(to solve the above problem(s), 1)*
Here I would check if `it.newStreams` is not empty instead of doing `it.newStreamsCount > 0`